nuxeo-drive-client/nxdrive/wui/conflicts.py

```diff
@@ -7,17 +7,102 @@
 from nxdrive.wui.dialog import WebDialog, WebDriveApi
 from nxdrive.wui.translator import Translator
 from PyQt4 import QtCore
+from Queue import PriorityQueue
+from time import sleep
+from datetime import timedelta, datetime
+from nxdrive.engine.workers import Worker
+
 
 log = get_logger(__name__)
 
 
+class UserNameResolver(Worker):
```

Comment:
Initially we considered one update to Users table per day. But this has a problem: If a conflict occurs after the Users table is updated on the day, then we need to retrieve the userid -- full user name relation again. So in addition to updating users every day, we need to do one of the following for new conflicts:
1. retrieving userid -- full user name on demand (like current implementation).
2. retrieve userid -- full user name as soon as sync happens.

May be we can change the current thread into a poll worker. Then make it capable of being triggered on demand if new conflict happens. We need to investigate and see if this approach will be less complex.